subprojects/core/src/main/java/org/gradle/internal/service/scopes/ProjectScopeServices.java

```diff
@@ -116,6 +117,7 @@ public class ProjectScopeServices implements ServiceRegistrationProvider {
 
     public static CloseableServiceRegistry create(
         ServiceRegistry parent,
+        SettingsInternal settings,
```

Comment:
🤔  We don't need the full Settings instance here. I suggest we use `org.gradle.initialization.layout.BuildLayout` instead (internal, not public interface). It should be [available](https://github.com/gradle/gradle/blob/d4571c1b57bbdfbc075dcc03f64f20680a7feb55/subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildScopeServices.java#L313-L314) as a Build-scoped service.